atlp-rwanda/ecommerce-app-predators-bn
# #29 Enable Product-Listing

#### What does this PR do?

- Adds the product listing routes for both vendors and buyers.
- A vendor can view all products in his collection
- A buyer can view all available products at time.
- Adds documentation of these routes in Swagger

#### Description of Task to be completed

> As a Seller I should be able to retrieve a list of all items in my collection
> As a buyer I should be able to retrieve a list of all items on the available list of products
> Such that a high level inspection is viable
> 
> Acceptance criteria
> 
> Given a get request to retrieve all of a seller's items
> Scenario 1
> When the requesting user/token is a seller and the target item already exist in the particular seller's collection and the request body is validated
> Then:
> A json object containing a status code, message, and items as an array should be returned. If no Items, an empty array be returned as the value of the items property
> Scenario 2
> When the request body fails any validation checks or the token is not of the role seller
> Then:
> 
> A json object containing an status property with the appropriate 4XX code, and an error property with the correct error message should be sent to the requesting client describing the validation error

#### How should this be manually tested?

- Make sure you have necessary environment variables (check the config directory)
- First checkout my branch `ft-product-listing-29`
- install necessary dependencies by running `npm install`
- start the server by running `npm run start`
- /!\ Note: All of the following routes are protected, you should access route `i`  with vendor privileges, and route `ii`  with buyer privileges.
- use postman to test these routes:
             **i**.  POST `http://localhost:3000/api/vendor/collection` with a requestBody containing id  property whose value is the id of the vendor requesting his collection.
             **ii.**  GET `http://localhost:3000/api/user/products`
#### What are the relevant pivotal tracker stories?

- [Card #29](https://trello.com/c/yBHTSZAj)
